### PR TITLE
On_boarding Actions and reducers

### PR DESCRIPTION
**Pull request by manavbp**

### CODE REVIEW

File: app/actions/helpers/launchpad.ts [hunk 1]
```diff
@@ -2,8 +2,8 @@ import { ipcRenderer } from 'electron';
 import AutoLaunch from 'auto-launch';
 
 import pkg from '$Package';
-import { UserPreferences } from '$Definitions/application.d';
-import { userPreferenceDatabase } from './user_preferences_db';
+import { Preferences } from '$Definitions/application.d';
+import { preferenceDatabase } from './preferences_db';
```
Comment (joshuef) [inline app/actions/helpers/launchpad.ts]: is this still a database? perhaps a name change?

Comment (manavbp) [reply app/actions/helpers/launchpad.ts]: Nope, it's not a database anymore, there should be a name change there! I'll update it right away.

File: app/actions/helpers/launchpad.ts [hunk 2]
```diff
@@ -12,27 +12,20 @@ export const mockPromise = ( data = null ) =>
...
+export const fetchPreferencesLocally = (): Promise<Preferences> =>
     new Promise( async ( resolve, reject ) => {
         try {
-            if ( !userPreferenceDatabase.isReady() ) {
-                await userPreferenceDatabase.init();
-            }
-            const [userPreferences] = await userPreferenceDatabase.getAll();
-            delete userPreferences.id;
-            return resolve( userPreferences );
+            const preferences = await settingsHandler.getPreferences();
```
Comment (joshuef) [inline app/actions/helpers/launchpad.ts]: why are we doing two layers of abstraction here? Why does the code need to be inside of a `settingsHandler`?

Comment (joshuef) [reply app/actions/helpers/launchpad.ts]: Kind of feels like this file doesnt need to exist and the action should just call `settingsHandler.getPreferences()` directly...

File: app/actions/helpers/settings_handler.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    defaultPreferences,
+    settingsHandlerName,
+    isRunningTestCafeProcess,
+    getAppFolderPath
+} from '$Constants/index';
+
+class SettingsHandler {
+    private preferenceId: number | null;
+
+    private tableName: string;
```
Comment (joshuef) [inline app/actions/helpers/settings_handler.ts]: we have no tables

Comment (joshuef) [reply app/actions/helpers/settings_handler.ts]: there is no db

File: app/actions/helpers/settings_handler.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+    public constructor() {
```
Comment (joshuef) [inline app/actions/helpers/settings_handler.ts]: why use a class?

File: app/constants/index.ts [hunk 5]
```diff
@@ -146,6 +146,13 @@ const safeNodeAppPath = () => {
     return nodeAppPath;
 };
 
+export const getAppFolderPath = () => {
+    if ( remote && remote.app ) {
+        return remote.app.getPath( 'appData' );
+    }
+    return null;
```
Comment (joshuef) [inline app/constants/index.ts]: we can get the path in the main process too...  https://electronjs.org/docs/api/app#appgetpathname

@manavbp `remote` is just a way to access the main process from renderer processes...

File: app/main.dev.ts [hunk 6]
```diff
@@ -64,6 +71,27 @@ const installExtensions = async () => {
     ).catch( console.log );
 };
 
+const createApplicationFolder = async () => {
```
Comment (joshuef) [inline app/main.dev.ts]: lets stop dumping code in `main`. We're allowed to create new files @manavbp . (those IPC funcs at the bottom should really be in a new file too)

File: app/main.dev.ts [hunk 7]
```diff
@@ -64,6 +71,27 @@ const installExtensions = async () => {
...
+    return new Promise( async ( resolve, reject ) => {
+        let appFolderPath = app.getPath( 'appData' );
+        const tableName = settingsHandlerName.production;
+
+        if ( !appFolderPath ) {
+            return reject( new Error( 'Unable to fetch application folder path' ) );
+        }
+        appFolderPath = path.resolve( appFolderPath, pkg.name );
+        try {
+            await fs.ensureDir( appFolderPath );
```
Comment (joshuef) [inline app/main.dev.ts]: I dont think we need this with `ensureFile` https://github.com/jprichardson/node-fs-extra/blob/HEAD/docs/ensureFile.md

File: app/main.dev.ts [hunk 8]
```diff
@@ -64,6 +71,27 @@ const installExtensions = async () => {
     ).catch( console.log );
 };
 
+const createApplicationFolder = async () => {
+    return new Promise( async ( resolve, reject ) => {
+        let appFolderPath = app.getPath( 'appData' );
+        const tableName = settingsHandlerName.production;
+
+        if ( !appFolderPath ) {
+            return reject( new Error( 'Unable to fetch application folder path' ) );
```
Comment (joshuef) [inline app/main.dev.ts]: or we make the folder?

File: package.json [hunk 9]
```diff
@@ -35,6 +35,7 @@
     "test": "cross-env NODE_ENV=test BABEL_DISABLE_CACHE=1 jest -u",
     "test-all": "yarn lint && yarn build && yarn test && yarn build-e2e && yarn test-e2e",
     "test-e2e": "node -r babel-register-ts ./internals/scripts/CheckBuiltsExist.js && cross-env NODE_ENV=test TEST_CAFE=true testcafe electron:. ./__e2e__/*.e2e.ts",
+    "settings": "node -r babel-register-ts ./internals/scripts/CheckBuiltsExist.js && cross-env NODE_ENV=test TEST_CAFE=true testcafe electron:. ./__e2e__/Settings.e2e.ts",
```
Comment (joshuef) [inline package.json]: why a command just for settings?

Comment (joshuef) [reply package.json]: (if it was convenience... you know you can test _only_ a specific test in testcafe w/ `test.only(...` just like jest?

**Pull request merged**